File: backend/app/schemas/alert.py
"""
backend/app/schemas/alert.py
Pydantic schemas for predictive high-risk alerts.
"""

from typing import Optional, List, Union
from datetime import datetime
from pydantic import BaseModel, Field, ConfigDict

class AlertResponse(BaseModel):
    id: int
    alert_id: str
    case_id: str
    prediction_id: str
    location_id: str
    location_name: Optional[str] = "ATM"
    bank_name: Optional[str] = "UNKNOWN"
    risk_score: float
    risk_level: str
    rank: int = 1
    message: Optional[str] = None
    status: str  # NEW, ACKNOWLEDGED, UNDER_REVIEW, RESOLVED
    acknowledged_by: Optional[str] = None
    acknowledged_at: Optional[Union[datetime, str]] = None
    created_at: Optional[Union[datetime, str]] = None

    model_config = ConfigDict(from_attributes=True)

class AlertAcknowledgeRequest(BaseModel):
    acknowledged_by: Optional[str] = Field(default="LEA_OFFICER_01", description="Officer ID or callsign")
    notes: Optional[str] = Field(default=None, description="Optional investigative action notes")

File: frontend/src/types/alert.ts
export interface Alert {
  id: number;
  alert_id: string;
  case_id: string;
  prediction_id: string;
  location_id: string;
  location_name?: string;
  bank_name?: string;
  risk_score: number;
  risk_level: 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW';
  rank: number;
  message?: string;
  status: 'NEW' | 'ACKNOWLEDGED' | 'UNDER_REVIEW' | 'RESOLVED';
  acknowledged_by?: string;
  acknowledged_at?: string;
  created_at?: string;
}
